File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/map/PolygonController.java
package com.valentingrigorean.arcgis_maps_flutter.map;

import android.graphics.Color;

import com.esri.arcgisruntime.mapping.view.Graphic;
import com.esri.arcgisruntime.symbology.SimpleFillSymbol;
import com.esri.arcgisruntime.symbology.SimpleLineSymbol;

public final class PolygonController extends BaseGraphicController {

    private final Graphic graphic;
    private final SimpleFillSymbol polygonSymbol;
    private final SimpleLineSymbol polygonStrokeSymbol;

    public PolygonController(String polygonId,SelectionPropertiesHandler selectionPropertiesHandler) {
        super(selectionPropertiesHandler);
        polygonStrokeSymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.BLACK, 10);
        polygonSymbol = new SimpleFillSymbol(SimpleFillSymbol.Style.SOLID, Color.BLACK, polygonStrokeSymbol);

        graphic = new Graphic();
        graphic.setSymbol(polygonSymbol);
        graphic.getAttributes().put("polygonId", polygonId);
    }

    public void setFillColor(int color) {
        polygonSymbol.setColor(color);
    }

    public void setStrokeColor(int color) {
        polygonStrokeSymbol.setColor(color);
    }

    public void setStrokeWidth(float width) {
        polygonStrokeSymbol.setWidth(width);
    }

    @Override
    protected Graphic getGraphic() {
        return graphic;
    }
}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/map/PolygonController.kt
package com.valentingrigorean.arcgis_maps_flutter.map

import android.graphics.Color
import com.arcgismaps.mapping.symbology.SimpleFillSymbol
import com.arcgismaps.mapping.symbology.SimpleLineSymbol
import com.arcgismaps.mapping.view.Graphic


class PolygonController(polygonId: String?) : BaseGraphicController(), PolygonControllerSink {
    protected override val graphic: Graphic
    private val polygonSymbol: SimpleFillSymbol
    private val polygonStrokeSymbol: SimpleLineSymbol

    init {
        polygonStrokeSymbol = SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.BLACK, 10f)
        polygonSymbol =
            SimpleFillSymbol(SimpleFillSymbol.Style.SOLID, Color.BLACK, polygonStrokeSymbol)
        graphic = Graphic()
        graphic.symbol = polygonSymbol
        graphic.attributes["polygonId"] = polygonId
    }

    override fun setFillColor(color: Int) {
        polygonSymbol.color = color
    }

    override fun setStrokeColor(color: Int) {
        polygonStrokeSymbol.color = color
    }

    override fun setStrokeWidth(width: Float) {
        polygonStrokeSymbol.width = width
    }

    override fun setStrokeStyle(style: SimpleLineSymbol.Style?) {
        polygonStrokeSymbol.style = style
    }
}